dotnet/src/SemanticKernel/Planning/SKContextExtensions.cs

```diff
@@ -2,47 +2,161 @@
+
+        List<FunctionView> result = availableFunctions;
+        if (!string.IsNullOrEmpty(semanticQuery))
+        {
+            // If a Memory provider has not been registered, do not filter the functions.
+            if (context.Memory is not NullMemory)
+            {
+                // catalog functions to memory
+                foreach (var function in availableFunctions)
+                {
+                    var functionName = function.ToFunctionName();
+                    var key = string.IsNullOrEmpty(function.Description) ? functionName : function.Description;
+
+                    // It'd be nice if there were a saveIfNotExists method on the memory interface
+                    var memoryEntry = await context.Memory.GetAsync(MemoryCollectionName, key, context.CancellationToken);
+                    if (memoryEntry == null)
+                    {
+                        // TODO It'd be nice if the minRelevanceScore could be a parameter for each item that was saved to memory
+                        // As folks may want to tune their functions to be more or less relevant.
+                        await context.Memory.SaveInformationAsync(MemoryCollectionName, key, functionName, function.ToManualString(),
+                            context.CancellationToken);
+                    }
+                }
+
+                var memories = context.Memory.SearchAsync(MemoryCollectionName, semanticQuery, config.MaxFunctions, config.RelevancyThreshold,
+                    context.CancellationToken);
+
+                result = new List<FunctionView>();
+                await foreach (var memoryEntry in memories)
+                {
+                    var function = availableFunctions.Find(x => x.ToFunctionName() == memoryEntry.Id);
+                    if (function != null)
+                    {
+                        result.Add(function);
+                    }
+                }
+
+                foreach (var function in includedFunctions)
+                {
+                    if (!result.Any(x => x.Name == function))
+                    {
+                        var functionView = availableFunctions.Find(x => x.Name == function);
+                        if (functionView != null)
+                        {
+                            result.Add(functionView);
+                        }
+                    }
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Gets the planner skill config from the SKContext.
+    /// </summary>
+    /// <param name="context">The SKContext to get the planner skill config from.</param>
+    /// <returns>The planner skill config.</returns>
+    internal static PlannerSkillConfig GetPlannerSkillConfig(this SKContext context)
+    {
+        var config = new PlannerSkillConfig();
+
+        if (context.Variables.Get(Parameters.RelevancyThreshold, out var threshold) && double.TryParse(threshold, out var parsedValue))
```

Comment:
>RelevancyThreshold [](http://example.com/codeflow?start=45&length=18)

Is there a way to customize this on a per function basis? In case we find that a particular function is being over/under used

Reply:
There's not right now, but I'm hopeful we can add that eventually. Using natural language in the goal text might be a good short term solution.